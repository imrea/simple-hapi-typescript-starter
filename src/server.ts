import Hapi from '@hapi/hapi';
import { PORT } from './config';

export const createServer = async () => {
  const server = Hapi.server({
    port: PORT,
  });

  server.route([
    {
      method: 'GET',
      path: '/',
      handler: (_request, _h) => {
        return {
          message: 'Hello Hapi!',
        };
      },
    },
  ]);

  return server;
};

export const initServer = async () => {
  const server = await createServer();
  await server.initialize();
  return server;
};

export const startServer = async () => {
  const server = await createServer();
  await server.start();
  return server;
};
