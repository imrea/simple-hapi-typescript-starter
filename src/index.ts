import { startServer } from './server';

async function main() {
  const server = await startServer();
  console.log(`Server running at: ${server.info.uri}`);
}

process.on('unhandledRejection', (err) => {
  console.error(err);
  process.exit(1);
});

main();
