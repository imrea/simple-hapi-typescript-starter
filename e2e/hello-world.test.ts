import Hapi from '@hapi/hapi';
import { createServer } from '../src/server';

let server: Hapi.Server;

beforeEach(async () => {
  server = await createServer();
  await server.initialize();
});

afterEach(async () => {
  await server.stop();
});

describe('GET /', () => {
  it('responds with 200', async () => {
    const res = await server.inject({
      method: 'GET',
      url: '/',
    });

    expect(res.statusCode).toBe(200);
    expect(res.result).toMatchSnapshot();
  });
});
