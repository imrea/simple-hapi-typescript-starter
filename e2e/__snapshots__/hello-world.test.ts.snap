// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET / responds with 200 1`] = `
Object {
  "message": "Hello World!",
}
`;
